crates/hermesllm/src/providers/response.rs

```diff
@@ -207,177 +346,191 @@ impl Into<Vec<u8>> for SseEvent {
                     transformed_event.sse_transform_buffer = format!("\n"); // suppress the event upstream for OpenAI
                 }
             }
+            _ => {
+                // Other combinations can be handled here as needed
+            }
         }
 
         Ok(transformed_event)
     }
 }
 
-#[derive(Debug)]
-pub struct SseParseError {
-    pub message: String,
+// TryFrom implementation to convert AWS Event Stream DecodedFrame to ProviderStreamResponseType
+impl TryFrom<(&aws_smithy_eventstream::frame::DecodedFrame, &SupportedAPIs, &SupportedUpstreamAPIs)> for ProviderStreamResponseType {
+    type Error = Box<dyn std::error::Error + Send + Sync>;
+
+    fn try_from((frame, client_api, upstream_api): (&aws_smithy_eventstream::frame::DecodedFrame, &SupportedAPIs, &SupportedUpstreamAPIs)) -> Result<Self, Self::Error> {
+        use aws_smithy_eventstream::frame::DecodedFrame;
+
+        match frame {
+            DecodedFrame::Complete(_) => {
+                // We have a complete frame - parse it based on upstream API
+                match (upstream_api, client_api) {
+                    (SupportedUpstreamAPIs::AmazonBedrockConverseStream(_), SupportedAPIs::AnthropicMessagesAPI(_)) => {
+                        // Parse the DecodedFrame into ConverseStreamEvent
+                        let bedrock_event = crate::apis::amazon_bedrock::ConverseStreamEvent::try_from(frame)?;
+                        let anthropic_event: crate::apis::anthropic::MessagesStreamEvent = bedrock_event.try_into()?;
+
+                        Ok(ProviderStreamResponseType::MessagesStreamEvent(anthropic_event))
+                    }
+                    (SupportedUpstreamAPIs::AmazonBedrockConverseStream(_), SupportedAPIs::OpenAIChatCompletions(_)) => {
+                        // Parse the DecodedFrame into ConverseStreamEvent
+                        let bedrock_event = crate::apis::amazon_bedrock::ConverseStreamEvent::try_from(frame)?;
+                        let openai_event: crate::apis::openai::ChatCompletionsStreamResponse = bedrock_event.try_into()?;
+                        Ok(ProviderStreamResponseType::ChatCompletionsStreamResponse(openai_event))
+                    }
+                    _ => {
+                        Err("Unsupported API combination for event-stream decoding".into())
+                    }
+                }
+            }
+            DecodedFrame::Incomplete => {
+                Err("Cannot convert incomplete frame to provider response".into())
+            }
+        }
+    }
 }
 
-impl fmt::Display for SseParseError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "SSE parse error: {}", self.message)
+
+/// AWS Event Stream frame decoder wrapper
+pub struct BedrockBinaryFrameDecoder<B>
+where
+    B: Buf,
+{
+    decoder: aws_smithy_eventstream::frame::MessageFrameDecoder,
+    buffer: B,
```

Comment:
```suggestion
    decoder: MessageFrameDecoder,
    buffer: B,
```

and import at file top aws_smithy_eventstream::frame::MessageFrameDecoder